### PR TITLE
Modal Footer

### PR DESCRIPTION
**Pull request by jonseitz**
Adds a Modal Footer component to the library, and integrates it into the Examples for the modal. Since the `ModalFooter` is just a `styled-component` wrapped in `withTheme`, I haven't added any tests for the component -- any tests would just be verifying that third-party libraries do what they're supposed to, and I don't think that's necessary for our needs. Happy to discuss further if you feel differently.

### CODE REVIEW

File: src/Modals/ModalFooter.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  /**
+   * the content of the Modal Footer
+   */
+  children: ReactNode;
+}
+
+
+const styledModalFooter = styled.div<ModalFooterProps>`
+  display: flex;
+  flex-direction: reverse-row;
```
Comment (natalynyu) [inline src/Modals/ModalFooter.tsx:14]: Do you mean `row-reverse` here? I'm not seeing a `reverse-row` option:
https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction
Also, could you clarify why we are ordering in reverse?

Comment (natalynyu) [reply src/Modals/ModalFooter.tsx:14]: From the example, it seems like we are ordering in `row` order

File: examples/Modals/ModalFooter.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+This is a utility component for adding a footer row to the `<Modal>`. It's set up to use `flexbox`, and fill in buttons from right to left. For example:
```
Comment (natalynyu) [inline examples/Modals/ModalFooter.md:1]: Should this be 'left to right'? If my comment from line 14 is wrong, you can disregard this

File: src/Modals/ModalFooter.tsx (hunk 1 above)
Comment (jonseitz) [reply src/Modals/ModalFooter.tsx:14]: Totally right, that should be `row-reverse` and I just did it wrong.

The reason for ordering in reverse is to follow the style of [Vittorio's mockups](https://github.com/seas-computing/course-planner/issues/169#issuecomment-601413769), where:
- **With only one button** the button is located in the bottom right corner
- **With two buttons** they're in the bottom right and bottom left corners (i.e., not right next to each other).
- I don't think we've specified what to do with 2+ buttons, but as of right now they'd be distributed evenly across the row.

The "reverse" part of it is less than ideal. We could use `justify-content: end;` with `flex-direction: row;`, but that would pack the items to the right, so the "cancel" button would be right next to the "confirm button" (not good). I'm open to other ideas though. Any thoughts?

File: examples/Modals/ModalFooter.md (hunk 2 above)
Comment (jonseitz) [reply examples/Modals/ModalFooter.md:1]: rationale for "right to left" above

File: src/Modals/ModalFooter.tsx (hunk 1 above)
Comment (natalynyu) [reply src/Modals/ModalFooter.tsx]: Oh I see what you're saying with the `reverse` part of this now. I think this is the best way to go

**Pull request merged**